crates/sui-types/src/nitro_attestation.rs

```diff
@@ -391,138 +400,215 @@ impl AttestationDocument {
+                for (k, v) in pairs.iter() {
+                    let key = k.as_integer().ok_or(
+                        NitroAttestationVerifyError::InvalidAttestationDoc(
+                            "invalid PCR key format".to_string(),
+                        ),
+                    )?;
+                    let value =
+                        v.as_bytes()
+                            .ok_or(NitroAttestationVerifyError::InvalidAttestationDoc(
+                                "invalid PCR value format".to_string(),
+                            ))?;
+
+                    if value.len() != 32 && value.len() != 48 && value.len() != 64 {
                         return Err(NitroAttestationVerifyError::InvalidAttestationDoc(
-                            "invalid PCR format".to_string(),
+                            "invalid PCR value length".to_string(),
                         ));
                     }
-                }
-                pcr_vec
-            }
-            _ => {
-                return Err(NitroAttestationVerifyError::InvalidAttestationDoc(
-                    "cannot parse PCRs".to_string(),
-                ))
-            }
-        };
 
-        let user_data = get_optional_bytes("user_data");
-        let nonce = get_optional_bytes("nonce");
+                    // Valid PCR indices are 0, 1, 2, 3, 4, 8 for AWS.
+                    let key_u64 = u64::try_from(key).map_err(|_| {
+                        NitroAttestationVerifyError::InvalidAttestationDoc(
+                            "invalid PCR index".to_string(),
+                        )
+                    })?;
+
+                    for i in [0, 1, 2, 3, 4, 8] {
+                        if key_u64 == i {
+                            pcr_vec.push(value.to_vec());
+                        }
+                    }
+                }
+                Ok(pcr_vec)
+            })?;
 
-        let cabundle = match document_map
-            .iter()
-            .find(|(k, _)| matches!(k, ciborium::value::Value::Text(s) if s == "cabundle"))
-        {
-            Some((_, ciborium::value::Value::Array(arr))) => arr
-                .iter()
-                .map(|v| match v {
-                    ciborium::value::Value::Bytes(bytes) => Ok(bytes.clone()),
-                    _ => Err(NitroAttestationVerifyError::InvalidAttestationDoc(
-                        "invalid cabundle format".to_string(),
-                    )),
-                })
-                .collect::<Result<Vec<_>, _>>()?,
-            _ => {
-                return Err(NitroAttestationVerifyError::InvalidAttestationDoc(
-                    "cannot parse cabundle".to_string(),
-                ))
-            }
-        };
+        let cabundle = document_map
+            .get("cabundle")
+            .ok_or(NitroAttestationVerifyError::InvalidAttestationDoc(
+                "cabundle not found".to_string(),
+            ))?
+            .as_array()
```

Comment:
can there be more than one cert chain?

Reply:
yes - only one chain that goes to the root of trust: https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html#chain

Reply:
let'd doc this and say that we assume the chains are in order